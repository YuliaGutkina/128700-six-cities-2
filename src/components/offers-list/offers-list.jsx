import React from 'react';
import PropTypes from "prop-types";
import {OfferCard} from "../offer-card/offer-card";


export const OffersList = (props) => {
  const {places, onCardTitleClick, onSetActive} = props;

  return <div className="cities__places-list places__list tabs__content">
    {places.map((offer) => <OfferCard
      key={offer.id}
      offer={offer}
      onTitleClick={onCardTitleClick}
      onHover={(item) => {
        onSetActive(item);
      }}
    />)}
  </div>;
};


OffersList.propTypes = {
  places: PropTypes.arrayOf(
      OfferCard.propTypes.offer
  ),
  onCardTitleClick: PropTypes.func.isRequired,
  onSetActive: PropTypes.func
};
