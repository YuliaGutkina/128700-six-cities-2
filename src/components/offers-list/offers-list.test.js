import React from "react";
import renderer from "react-test-renderer";
import {OffersList} from "./offers-list";


it(`Offers list correctly renders after relaunch`, () => {
  const tree = renderer
    .create(
        <OffersList
          places={[{}, {}]}
          onCardTitleClick={jest.fn()}
          onSetActive={jest.fn()}
        />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
