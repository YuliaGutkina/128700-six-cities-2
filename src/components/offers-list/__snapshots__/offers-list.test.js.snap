// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers list correctly renders after relaunch 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <withRouter(Connect(OfferCard))
    offer={Object {}}
    onHover={[Function]}
    onTitleClick={[MockFunction]}
  />
  <withRouter(Connect(OfferCard))
    offer={Object {}}
    onHover={[Function]}
    onTitleClick={[MockFunction]}
  />
</div>
`;
